### PR TITLE
Phpmyadmin chart

### PR DESCRIPTION
**Pull request by vtuson**
**What this PR does / why we need it**:
phpMyAdmin is a free and open source administration tool for MySQL and MariaDB. As a portable web application written primarily in PHP, it has become one of the most popular MySQL administration tools, especially for web hosting services

Adds a new Chart that can be used standalone or as a dependency in charts that use mysql or mariadb to admin the database. 

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by vtuson** [comment]
I have asked a few people to review before I assign it to be merged

File: stable/phpmyadmin/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+name: phpmyadmin
+version: 0.1.0
+appVersion: 4.8.0
+description: PhpMyAdmin is an mysql admininstration frontend
```
Comment (juan131) [inline stable/phpmyadmin/Chart.yaml]: Typo: `administration`
Use `MySQL`
I see the same description on other charts maintained by Bitnami on _Chart.yaml_ and the associated Docker Image repository. I'd use the same approach here

File: stable/phpmyadmin/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+keywords:
```
Comment (juan131) [inline stable/phpmyadmin/Chart.yaml]: I'd add `php` to keywords

File: stable/phpmyadmin/Chart.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+- mariadb
+- mysql
+- phpmyadmin
+home: https://www.phpmyadmin.net/
+icon: https://upload.wikimedia.org/wikipedia/commons/thumb/4/4f/PhpMyAdmin_logo.svg/800px-PhpMyAdmin_logo.svg.png
+sources:
+- https://github.com/bitnami/bitnami-docker-phpmyadmin
+maintainers:
+- name: Bitnami
```
Comment (juan131) [inline stable/phpmyadmin/Chart.yaml]: I think you need to use `bitnami-bot` here

File: stable/phpmyadmin/README.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
+# PhpMyAdmin
```
Comment (juan131) [inline stable/phpmyadmin/README.md]: I noticed you use `PhpMyAdmin` sometimes and `Phpmyadmin` other times. Try be consistent on this.

File: stable/phpmyadmin/README.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+## Uninstalling the Chart
+
+To uninstall/delete the `my-release` deployment:
+
+```console
+$ helm delete my-release
+```
+
+The command removes all the Kubernetes components associated with the chart and deletes the release.
```
Comment (juan131) [inline stable/phpmyadmin/README.md:39]: Suggestion:

The command removes all the Kubernetes components associated with the chart and deletes the release. Remove also the release from the store using `--purge` option:

```console
$ helm delete --purge my-release
```

File: stable/phpmyadmin/templates/NOTES.txt [hunk 6]
```diff
@@ -0,0 +1,34 @@
+1. Get the application URL by running these commands:
```
Comment (juan131) [inline stable/phpmyadmin/templates/NOTES.txt:2]: Add some warning at the top:

```
** Please be patient while the chart is being deployed **
```

File: stable/phpmyadmin/templates/NOTES.txt [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+           You can watch the status of by running 'kubectl get svc -w {{ template "phpmyadmin.fullname" . }}'
+  export SERVICE_IP=$(kubectl get svc --namespace {{ .Release.Namespace }} {{ template "phpmyadmin.fullname" . }} -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+  echo http://$SERVICE_IP:{{ .Values.service.port }}
+{{- else if contains "ClusterIP" .Values.phpmyadmin.service_type }}
+  export POD_NAME=$(kubectl get pods --namespace {{ .Release.Namespace }} -l "app={{ template "phpmyadmin.name" . }},release={{ .Release.Name }}" -o jsonpath="{.items[0].metadata.name}")
+  echo "Visit http://127.0.0.1:8080 to use your application"
+  kubectl port-forward $POD_NAME 8080:80
+{{- end }}
+
+##### HOW TO LOG IN
```
Comment (juan131) [inline stable/phpmyadmin/templates/NOTES.txt]: I'd simply use `PhpMyAdmin credentials:`

File: stable/phpmyadmin/templates/deployment.yaml [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+    spec:
+      {{- if .Values.image.pullSecrets }}
+      imagePullSecrets:
+      {{- range .Values.image.pullSecrets }}
+        - name: {{ . }}
+      {{- end}}
+      {{- end }}
+      containers:
+        - name: {{ .Chart.Name }}
+          image: "{{ .Values.image.registry }}/{{ .Values.image.repository }}:{{ .Values.image.tag }}"
```
Comment (juan131) [inline stable/phpmyadmin/templates/deployment.yaml]: Define a "template" function on *_helpers.tpl* for this:

```
{{/*
Return the proper image name
*/}}
{{- define "phpmyadmin.image" -}}
{{- $registryName :=  .Values.image.registry -}}
{{- $repositoryName := .Values.image.repository -}}
{{- $tag := .Values.image.tag -}}
{{- printf "%s/%s:%s" $registryName $repositoryName $tag -}}
{{- end -}}
```

Then use:

```
image: "{{ template "phpmyadmin.image" . }}"
```

File: stable/phpmyadmin/templates/service.yaml [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+kind: Service
+metadata:
+  name: {{ template "phpmyadmin.fullname" . }}
+  labels:
+    app: {{ template "phpmyadmin.name" . }}
+    chart: {{ template "phpmyadmin.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+spec:
+  type: {{ .Values.phpmyadmin.service_type }}
```
Comment (juan131) [inline stable/phpmyadmin/templates/service.yaml]: Bitnami is using `service.type` on other solutions

File: stable/phpmyadmin/templates/service.yaml [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+  ports:
+    - port: {{ .Values.phpmyadmin.service_port }}
```
Comment (juan131) [inline stable/phpmyadmin/templates/service.yaml]: Bitnami is using `service.port` on other solutions

File: stable/phpmyadmin/values.yaml [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+  ##
+  # pullSecrets:
+  #   - myRegistrKeySecretName
+
+## User of the application
+## ref: https://github.com/bitnami/bitnami-docker-wordpress#environment-variables
+##
+
+phpmyadmin:
+  service_type: ClusterIP
```
Comment (juan131) [inline stable/phpmyadmin/values.yaml]: Bitnami is using `service.type` &  `service.port` on other solutions:

```
phpmyadmin:
  service:
     type: ClusterIP
     port: 80
...
```

**Review by juan131 — CHANGES_REQUESTED** [review]

File: stable/phpmyadmin/values.yaml (hunk 11 above)
Comment (vtuson) [reply stable/phpmyadmin/values.yaml]: I dont think having another extra layer in the yaml is justifyied for 2 options

File: stable/phpmyadmin/templates/service.yaml (hunk 10 above)
Comment (vtuson) [reply stable/phpmyadmin/templates/service.yaml]: I dont think having another extra layer in the yaml is justifyied for 2 options

File: stable/phpmyadmin/templates/service.yaml (hunk 9 above)
Comment (vtuson) [reply stable/phpmyadmin/templates/service.yaml]: I dont think having another extra layer in the yaml is justifyied for 2 options

File: stable/phpmyadmin/templates/NOTES.txt (hunk 7 above)
Comment (vtuson) [reply stable/phpmyadmin/templates/NOTES.txt]: you have to pass additional information if you have not configure a db host

File: stable/phpmyadmin/README.md (hunk 5 above)
Comment (vtuson) [reply stable/phpmyadmin/README.md:39]: yes, but also deletes the history, I dont think we want to suggest that as default option

File: stable/phpmyadmin/templates/deployment.yaml (hunk 8 above)
Comment (vtuson) [reply stable/phpmyadmin/templates/deployment.yaml]: ok

File: stable/phpmyadmin/templates/NOTES.txt [hunk 12]
```diff
@@ -0,0 +1,34 @@
+1. Get the application URL by running these commands:
+{{- if .Values.ingress.enabled }}
```
Comment (vtuson) [inline stable/phpmyadmin/templates/NOTES.txt:3]: I think the confusion comes from the hearders, I have fix that

File: stable/phpmyadmin/templates/NOTES.txt (hunk 6 above)
Comment (vtuson) [reply stable/phpmyadmin/templates/NOTES.txt:2]: since the message scrolls up, I have put this message at the bottom, so the user will see it regardless of the terminal size used

File: stable/phpmyadmin/README.md (hunk 4 above)
Comment (vtuson) [reply stable/phpmyadmin/README.md]: ok

File: stable/phpmyadmin/values.yaml [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+  ##
+  # pullSecrets:
+  #   - myRegistrKeySecretName
+
+## User of the application
+## ref: https://github.com/bitnami/bitnami-docker-wordpress#environment-variables
+##
+
+phpmyadmin:
+  serviceType: ClusterIP
```
Comment (juan131) [inline stable/phpmyadmin/values.yaml]: Better using camelCase! Thanks

File: stable/phpmyadmin/templates/NOTES.txt (hunk 6 above)
Comment (juan131) [reply stable/phpmyadmin/templates/NOTES.txt:2]: Did you miss this?

File: stable/phpmyadmin/templates/_helpers.tpl [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
+
+{{/*
+Create a default fully qualified app name.
+We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+*/}}
+{{- define "db.fullname" -}}
+{{- printf "%s-%s" .Release.Name .Values.phpmyadmin.dbSuffix | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+{{/*
+Return the proper image name
```
Comment (juan131) [inline stable/phpmyadmin/templates/_helpers.tpl]: Great!

File: stable/phpmyadmin/README.md (hunk 5 above)
Comment (juan131) [reply stable/phpmyadmin/README.md:39]: not done?

File: stable/phpmyadmin/Chart.yaml (hunk 1 above)
Comment (juan131) [reply stable/phpmyadmin/Chart.yaml]: not done?

File: stable/phpmyadmin/Chart.yaml (hunk 2 above)
Comment (juan131) [reply stable/phpmyadmin/Chart.yaml]: not done?

**Review by juan131 — COMMENTED** [review]
Thanks for the improvements @vtuson !!

You missed some of my comments. Take a look when you have a chance

File: stable/phpmyadmin/templates/NOTES.txt (hunk 6 above)
Comment (vtuson) [reply stable/phpmyadmin/templates/NOTES.txt:2]: I have implemented at the botton

File: stable/phpmyadmin/README.md (hunk 5 above)
Comment (vtuson) [reply stable/phpmyadmin/README.md:39]: nope, I dont think this should be in the chart tips

File: stable/phpmyadmin/Chart.yaml (hunk 2 above)
Comment (vtuson) [reply stable/phpmyadmin/Chart.yaml]: I dont think it is relevant

File: stable/phpmyadmin/Chart.yaml (hunk 1 above)
Comment (vtuson) [reply stable/phpmyadmin/Chart.yaml]: oops, done now

File: stable/phpmyadmin/Chart.yaml (hunk 3 above)
Comment (vtuson) [reply stable/phpmyadmin/Chart.yaml]: done

**Review by juan131 — APPROVED** [review]

**Comment by vtuson** [comment]
/assign @mattfarina 

suggested by @carrodher hope that is ok with you @mattfarina

File: stable/phpmyadmin/README.md [hunk 15]
```diff
@@ -0,0 +1,80 @@
+# PhpMyAdmin
+
+[Phpmyadmin](https://www.phpmyadmin.net/) is a free and open source administration tool for MySQL and MariaDB. As a portable web application written primarily in PHP, it has become one of the most popular MySQL administration tools, especially for web hosting services.
+
+## TL;DR;
```
Comment (sameersbn) [inline stable/phpmyadmin/README.md]: -> `TL;DR`

File: stable/phpmyadmin/README.md [hunk 16]
```diff
@@ -0,0 +1,80 @@
...
+## Configuration
+
+The following table lists the configurable parameters of the Phpmyadmin chart and their default values.
+
+|              Parameter               |               Description                |                         Default                         |
+|--------------------------------------|------------------------------------------|---------------------------------------------------------|
+| `image.registry`                     | PhpMyAdmin image registry                 | `docker.io`                                             |
+| `image.repository`                   | PhpMyAdmin Image name                     | `bitnami/phpmyadmin`                                     |
+| `image.tag`                          | PhpMyAdmin Image tag                      | `{VERSION}`                                             |
+| `image.pullPolicy`                   | Image pull policy                        | `Always` if `imageTag` is `latest`, else `IfNotPresent` |
```
Comment (sameersbn) [inline stable/phpmyadmin/README.md]: The default value column should just say `IfNotPresent`

File: stable/phpmyadmin/README.md [hunk 17]
```diff
@@ -0,0 +1,80 @@
...
+| `image.pullSecrets`                  | Specify image pull secrets               | `nil`                                                   |
+| `phpmyadmin.serviceType`            | type of service for PhpMyAdmin frontend             | `ClusterIP`                                                  |
+| `phpmyadmin.servicePort`        | port to expose service                   | `80`                                                   |
+| `phpmyadmin.dbPort`            | database port to use to connect                  | `3360`                                     |
+| `phpmyadmin.dbSuffix`                | Database suffix if included in the same release                  | `nil`                                          |
+| `phpmyadmin.dbHost`            | database host to connect to               | `nil`          |
```
Comment (sameersbn) [inline stable/phpmyadmin/README.md]: Are you sure we should be using a top-level `phpmyadmin` key for these params? Typically we use a structure such as the following in the values.yaml:

```
service:
  type: ClusterIP
  port: 80

db:
  host: 
  port: 3360
  suffix: 
```

File: stable/phpmyadmin/README.md [hunk 18]
```diff
@@ -0,0 +1,80 @@
...
+| `ingress.annotations`            | ingress annotations              | `{}`          |
+| `ingress.path`            | path to access frontend               | `/`          |
+| `ingress.host`            | ingress host               | `nil`          |
+| `ingress.tls`            | tls for ingress               | `[]`          |
+| `resources`                          | CPU/Memory resource requests/limits      | `{}`      |
+
+The above parameters map to the env variables defined in [bitnami/phpmyadmin](http://github.com/bitnami/bitnami-docker-Phpmyadmin). For more information please refer to the [bitnami/phpmyadmin](http://github.com/bitnami/bitnami-docker-Phpmyadmin) image documentation.
```
Comment (sameersbn) [inline stable/phpmyadmin/README.md]: Most of the parameters configure the kubernetes resource deployments. I think we should remove this line which says that the parameters map to env variables of the image.

File: stable/phpmyadmin/values.yaml [hunk 19]
```diff
@@ -0,0 +1,58 @@
+## Bitnami WordPress image version
+## ref: https://hub.docker.com/r/bitnami/phpmyadmin/tags/
+##
+image:
+  registry: docker.io
+  repository: bitnami/phpmyadmin
+  tag: 4.8.0
+  ## Specify a imagePullPolicy
+  ## Defaults to 'Always' if image tag is 'latest', else set to 'IfNotPresent'
```
Comment (sameersbn) [inline stable/phpmyadmin/values.yaml]: IMO this comment should be removed.

File: stable/phpmyadmin/templates/deployment.yaml [hunk 20]
```diff
@@ -0,0 +1,67 @@
...
+              containerPort: 443
+              protocol: TCP
+          livenessProbe:
+            httpGet:
+              path: /
+              port: http
+          readinessProbe:
+            httpGet:
+              path: /
+              port: http
```
Comment (sameersbn) [inline stable/phpmyadmin/templates/deployment.yaml:66]: would be nice to be able to enable/disable the  probes?

**Review by sameersbn — CHANGES_REQUESTED** [review]

File: stable/phpmyadmin/README.md [hunk 21]
```diff
@@ -0,0 +1,80 @@
...
+$ helm install stable/phpmyadmin
+```
+
+## Introduction
+
+This chart bootstraps a [Phpmyadmin](https://github.com/bitnami/bitnami-docker-phpmyadmin) deployment on a [Kubernetes](http://kubernetes.io) cluster using the [Helm](https://helm.sh) package manager.
+
+## Prerequisites
+
+- Kubernetes 1.4+ with Beta APIs enabled
```
Comment (unguiculus) [inline stable/phpmyadmin/README.md]: 1.8+

File: stable/phpmyadmin/templates/_helpers.tpl [hunk 22]
```diff
@@ -0,0 +1,49 @@
...
+
+{{/*
+Create chart name and version as used by the chart label.
+*/}}
+{{- define "phpmyadmin.chart" -}}
+{{- printf "%s-%s" .Chart.Name .Chart.Version | replace "+" "_" | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+
+{{/*
+Create a default fully qualified app name.
```
Comment (unguiculus) [inline stable/phpmyadmin/templates/_helpers.tpl]: app name?

File: stable/phpmyadmin/templates/_helpers.tpl [hunk 23]
```diff
@@ -0,0 +1,49 @@
...
+We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+*/}}
+{{- define "db.fullname" -}}
```
Comment (unguiculus) [inline stable/phpmyadmin/templates/_helpers.tpl]: All partials should be prefixed with the chart name, thus: `phpmyadmin.db.fullname`. Also, you should base it on the standard fullname template. Truncating as done here is probably not useful either because you'd strip off the db suffix. I'd just go with this:

`{{- template "phpmyadmin.fullname" . }}-{{ .Values.phpmyadmin.dbSuffix -}}`

But then I don't understand why this is needed or would work at all. In comments in `values.yaml` you mention this can be used if the DB is deployed as part of a release. In order for this to be possible there would need to be a dependency to a DB chart, but this chart doesn't have any dependencies.

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: stable/phpmyadmin/templates/_helpers.tpl (hunk 23 above)
Comment (vtuson) [reply stable/phpmyadmin/templates/_helpers.tpl]: Hi @unguiculus, maybe I am doing something wrong here. The usecase I am trying to cater for is to include phpmyadmin is as a dependency itself. For example Wordpress. Wordpress chart has mariadb as a dependency. Now, lets say that we add phpmyadmin also.

Wordpress is going to have:
releasename-wordpress
releasename-mariadb
releasename-phpmyadmin

some how when I do helm install wordpress, I need to pass in the values.yaml a name for phpmyadmin to point to. If that name is fixed, I can just use phpmyadmin.dbHost , if it is not (like in this case) i can set phpmyadmin.dbSuffix = mariadb and would try to connect to releasename-mariadb. 

I there a better way to achieve this?

File: stable/phpmyadmin/README.md (hunk 21 above)
Comment (vtuson) [reply stable/phpmyadmin/README.md]: thanks, copy/paste error 😱

File: stable/phpmyadmin/templates/_helpers.tpl (hunk 22 above)
Comment (vtuson) [reply stable/phpmyadmin/templates/_helpers.tpl]: I agree, makes no sense. Edited.

File: stable/phpmyadmin/README.md (hunk 15 above)
Comment (vtuson) [reply stable/phpmyadmin/README.md]: seems like additional ; is largely used

https://github.com/kubernetes/charts/search?utf8=%E2%9C%93&q=tl%3Bdr%3B&type=
https://github.com/kubernetes/charts/tree/master/stable/wordpress#tldr

If this is a blocker to land it I am happy to change it.

File: stable/phpmyadmin/README.md (hunk 16 above)
Comment (vtuson) [reply stable/phpmyadmin/README.md]: done

File: stable/phpmyadmin/README.md (hunk 17 above)
Comment (vtuson) [reply stable/phpmyadmin/README.md]: I think because it is likely that this chart will be used as a dependency, it think your comment makes sense. I will change it

File: stable/phpmyadmin/README.md (hunk 18 above)
Comment (vtuson) [reply stable/phpmyadmin/README.md]: thanks

File: stable/phpmyadmin/values.yaml (hunk 19 above)
Comment (vtuson) [reply stable/phpmyadmin/values.yaml]: yeap

File: stable/phpmyadmin/templates/deployment.yaml (hunk 20 above)
Comment (vtuson) [reply stable/phpmyadmin/templates/deployment.yaml:66]: it wont hurt I guess

**Comment by vtuson** [comment]
@sameersbn thanks for your feedback, I have submitted changes accordingly

File: stable/phpmyadmin/README.md (hunk 15 above)
Comment (sameersbn) [reply stable/phpmyadmin/README.md]: Well, I have to admit that I am responsible for the widespread use of `Tl;DR;` in this repo. You see, I used `TL;DR;` in the README's of the very first charts in this repo. As the project grew other contributors used the README as a template for their contributions which made this error more widespread.

Recently I realized that `TL;DR` is the correct usage (https://en.wikipedia.org/wiki/TL;DR) and I am only trying to undo the damage that i've already done :stuck_out_tongue: 

TL;DR: It's not a blocker :upside_down_face:

File: stable/phpmyadmin/README.md (hunk 4 above)
Comment (sameersbn) [reply stable/phpmyadmin/README.md]: `phpMyAdmin` is correct way to write it (as seen on https://www.phpmyadmin.net/)

File: stable/phpmyadmin/templates/NOTES.txt [hunk 24]
```diff
@@ -0,0 +1,36 @@
...
+{{if .Values.db.port}}with port {{.Values.db.port}} {{end }}
+Please login using a database username and password. 
+{{ else }}
+PhpMyAdmin has not been configure to point to a specific database. Please provide the db host, username and password at log in or upgrade the release with 
+a specific database
+$ helm upgrade {{.Release.Name}} stable/phpmyadmin --set db.host=mydb 
+{{end}}
+
+** Please be patient while the chart is being deployed **
+
```
Comment (sameersbn) [inline stable/phpmyadmin/templates/NOTES.txt]: file could use some formatting to make things more clear/readable

File: stable/phpmyadmin/templates/_helpers.tpl (hunk 23 above)
Comment (prydonius) [reply stable/phpmyadmin/templates/_helpers.tpl]: I think this is fine, I agree with renaming this to `phpmyadmin.db.fullname` or something similar.

File: stable/phpmyadmin/templates/NOTES.txt (hunk 24 above)
Comment (vtuson) [reply stable/phpmyadmin/templates/NOTES.txt]: done

File: stable/phpmyadmin/README.md (hunk 4 above)
Comment (sameersbn) [reply stable/phpmyadmin/README.md]: @vtuson as @juan131 suggested, we should be consistent with the application name (`phpMyAdmin`) in the README and other places.

File: stable/phpmyadmin/templates/NOTES.txt [hunk 25]
```diff
@@ -0,0 +1,50 @@
...
+2. How to log in
+{{ if or (.Values.db.suffix) (.Values.db.host)}}
+
+PhpMyAdmin has been configured to connect to a database in {{ if .Values.db.suffix}}{{template "phpmyadmin.dbfullname" .}}{{ else }}{{.Values.db.host}}{{end}}
+{{if .Values.db.port}}with port {{.Values.db.port}} {{end }}
+Please login using a database username and password. 
+
+{{ else }}
+PhpMyAdmin has not been configure to point to a specific database. Please provide the db host, username and password at log in or upgrade the release with
```
Comment (sameersbn) [inline stable/phpmyadmin/templates/NOTES.txt]: Please consider breaking this line (ideally wrap at 80 columns)

File: stable/phpmyadmin/README.md [hunk 26]
```diff
@@ -0,0 +1,80 @@
...
+| `image.pullSecrets`                  | Specify image pull secrets               | `nil`                                                   |
+| `service.type`            | type of service for PhpMyAdmin frontend             | `ClusterIP`                                                  |
+| `service.port`        | port to expose service                   | `80`                                                   |
+| `db.port`            | database port to use to connect                  | `3360`                                     |
+| `db.suffix`                | Database suffix if included in the same release                  | `nil`                                          |
+| `db.host`            | database host to connect to               | `nil`          |
+| `ingress.annotations`            | ingress annotations              | `{}`          |
+| `ingress.path`            | path to access frontend               | `/`          |
+| `ingress.host`            | ingress host               | `nil`          |
+| `ingress.tls`            | tls for ingress               | `[]`          |
```
Comment (sameersbn) [inline stable/phpmyadmin/README.md]: `ingress.enabled` is not documented

File: stable/phpmyadmin/templates/ingress.yaml [hunk 27]
```diff
@@ -0,0 +1,41 @@
+{{- if .Values.ingress.enabled -}}
```
Comment (sameersbn) [inline stable/phpmyadmin/templates/ingress.yaml:1]: I wasn't able to get the chart to deploy with `--set ingress.enabled=true,ingress.host=phpmyadmin.example.com`. 

```
Error: YAML parse error on phpmyadmin/templates/ingress.yaml: error converting YAML to JSON: yaml: line 12: mapping values are not allowed in this context
```

I modified the ingress.yaml as described below and deploy successfully with `ingress.enabled=true,ingress.hosts[0]=phpmyadmin.example.com`

### templates/ingress.yaml

```yaml
{{- if .Values.ingress.enabled -}}
{{- $fullName := include "phpmyadmin.fullname" . -}}
{{- $servicePort := .Values.service.port -}}
{{- $ingressPath := .Values.ingress.path -}}
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: {{ $fullName }}
  labels:
    app: {{ template "phpmyadmin.name" . }}
    chart: {{ template "phpmyadmin.chart" . }}
    release: {{ .Release.Name }}
    heritage: {{ .Release.Service }}
  annotations:
    {{- range $key, $value := .Values.ingress.annotations }}
      {{ $key }}: {{ $value | quote }}
    {{- end }}
spec:
  backend:
    serviceName: {{ $fullName }}
    servicePort: {{ $servicePort }}
{{- if .Values.ingress.hosts }}
  rules:
  {{- range $host := .Values.ingress.hosts }}
  - host: {{ $host }}
    http:
      paths:
        - path: {{ $ingressPath }}
          backend:
            serviceName: {{ $fullName }}
            servicePort: {{ $servicePort }}
  {{- end -}}
{{- end -}}
  {{- if .Values.ingress.tls }}
  tls:
{{ toYaml .Values.ingress.tls | indent 4 }}
  {{- end -}}
{{- end -}}
```

### values.yaml

```yaml
ingress:
  enabled: false
  annotations: {}
    # kubernetes.io/ingress.class: nginx
    # kubernetes.io/tls-acme: "true"
  ## path you want to map the phpmyadmin interface to
  path: /
  hosts:
    - phpmyadmin.local
  tls: []
  #  - secretName: chart-example-tls
  #    hosts:
  #      - chart-example.local
```


### templates/NOTES.txt (bonus)

```text

1. Get the application URL by running these commands:

{{- if .Values.ingress.enabled }}
{{ range .Values.ingress.hosts }}
  - http://{{ . }}
{{- end }}

{{- else if contains "NodePort" .Values.service.type }}
  export NODE_PORT=$(kubectl get --namespace {{ .Release.Namespace }} -o jsonpath="{.spec.ports[0].nodePort}" services {{ template "phpmyadmin.fullname" . }})
  export NODE_IP=$(kubectl get nodes --namespace {{ .Release.Namespace }} -o jsonpath="{.items[0].status.addresses[0].address}")
  echo http://$NODE_IP:$NODE_PORT

{{- else if contains "LoadBalancer" .Values.service.type }}

  NOTE: It may take a few minutes for the LoadBalancer IP to be available.
         You can watch the status of by running 'kubectl get svc -w {{ template "phpmyadmin.fullname" . }}'

  export SERVICE_IP=$(kubectl get svc --namespace {{ .Release.Namespace }} {{ template "phpmyadmin.fullname" . }} -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
  echo http://$SERVICE_IP:{{ .Values.service.port }}

{{- else if contains "ClusterIP" .Values.service.type }}

  export POD_NAME=$(kubectl get pods --namespace {{ .Release.Namespace }} -l "app={{ template "phpmyadmin.name" . }},release={{ .Release.Name }}" -o jsonpath="{.items[0].metadata.name}")

  Visit http://127.0.0.1:8080 to use your application

  kubectl port-forward $POD_NAME 8080:80

{{- end }}

2. How to log in

{{- if or (.Values.db.suffix) (.Values.db.host) }}

phpMyAdmin has been configured to connect to a database in {{ if .Values.db.suffix}}{{template "phpmyadmin.dbfullname" .}}{{ else }}{{.Values.db.host}}{{end}}
{{if .Values.db.port}}with port {{.Values.db.port}}{{end }}. Please login using a database username and password.

{{- else }}

phpMyAdmin has not been configure to point to a specific database.
Please provide the db host, username and password at log in or
upgrade the release with a specific database:

  helm upgrade {{.Release.Name}} stable/phpmyadmin --set db.host=mydb

{{- end }}

** Please be patient while the chart is being deployed **
```

**Review by sameersbn — CHANGES_REQUESTED** [review]

File: stable/phpmyadmin/templates/ingress.yaml (hunk 27 above)
Comment (vtuson) [reply stable/phpmyadmin/templates/ingress.yaml:1]: I will have a look at it tomorrow, i dont think that needs to have support for multiple hosts, something must have gone wrong in the process of removing that

Comment (sameersbn) [reply stable/phpmyadmin/templates/ingress.yaml:1]: yea sure, there isn't a need to support multiple hosts. We just need to drop the `{{- range $host := .Values.ingress.hosts }}` loop to remove it.

File: stable/phpmyadmin/templates/_helpers.tpl (hunk 23 above)
Comment (sameersbn) [reply stable/phpmyadmin/templates/_helpers.tpl]: This is a really cool feature, but I think the `suffix` parameter name is confusing/vague.
I would suggest renaming it something along the lines of `sqlServiceName`.

File: stable/phpmyadmin/README.md (hunk 26 above)
Comment (vtuson) [reply stable/phpmyadmin/README.md]: added

File: stable/phpmyadmin/templates/NOTES.txt (hunk 25 above)
Comment (vtuson) [reply stable/phpmyadmin/templates/NOTES.txt]: fixed

File: stable/phpmyadmin/templates/ingress.yaml (hunk 27 above)
Comment (vtuson) [reply stable/phpmyadmin/templates/ingress.yaml:1]: i turn out that I had 2 extra spaces in the ingress template for http. Working now

**Review by sameersbn — APPROVED** [review]

File: stable/phpmyadmin/templates/_helpers.tpl (hunk 23 above)
Comment (vtuson) [reply stable/phpmyadmin/templates/_helpers.tpl]: the only thing is that is more like the chart name than the full service name, maybe sqlChartName?

Comment (vtuson) [reply stable/phpmyadmin/templates/_helpers.tpl]: changed to db.chartName

**Comment by sameersbn** [comment]
ping @unguiculus
